src/js/providers/tracks-mixin.ts

```diff
@@ -310,10 +310,16 @@ const Tracks: TracksMixin = {
 
             for (i; i < len; i++) {
                 const track = tracks[i] as TextTrackLike;
-                let trackId: string = track._id || '';
-                if (!trackId) {
+                let trackId: string = track._id || '';                
+
+                if (!trackId) {   
+                    if (track.inuse === false && track.kind === 'captions' && this.renderNatively) {
+                        // ignore native captions tracks from previous items that no longer can be re-used
+                        track._id = createId(track, i)
+                        continue;
+                    }
                     if (track.kind === 'captions' || track.kind === 'metadata') {
-                        trackId = track._id = 'native' + track.kind + i;
+                        trackId = track._id = createId(track, i);
```

Comment:
this line is not part of what we're trying to fix and should not be changed.

Reply:
If we need to do `'native' + track.kind + i` in two places make a new function to make that DRY. Do not overload or use `createId` where not intended.